esp-hal/src/spi/master.rs

```diff
@@ -425,92 +417,14 @@ impl Address {
     }
 }
 
-/// Read and Write in half duplex mode.
-pub trait HalfDuplexReadWrite {
-    /// The associated error type that will be returned in the event of a
-    /// failure.
-    type Error;
-
-    /// Half-duplex read.
-    fn read(
-        &mut self,
-        data_mode: SpiDataMode,
-        cmd: Command,
-        address: Address,
-        dummy: u8,
-        buffer: &mut [u8],
-    ) -> Result<(), Self::Error>;
-
-    /// Half-duplex write.
-    fn write(
-        &mut self,
-        data_mode: SpiDataMode,
-        cmd: Command,
-        address: Address,
-        dummy: u8,
-        buffer: &[u8],
-    ) -> Result<(), Self::Error>;
-}
-
 /// SPI peripheral driver
-pub struct Spi<'d, M, T = AnySpi> {
+pub struct Spi<'d, T = AnySpi> {
```

Comment:
Given that we're messing around with these types already, does it make sense to add the `Async|Blocking` param to the basic `Spi` instance here in this PR?

Reply:
Isn't this big enough already? 🤣